php/libraries/Utility.class.inc

```diff
@@ -916,5 +916,28 @@ class Utility
         }
         return $languages;
     }
+
+    /**
+     * Takes an argument and determine whether it is a positive integer
+     * i.e. > 0.
+     *
+     * @param mixed $value The value to check. Intentionally not type hinted as
+     * this function is meant to assist in cases where the type is unknown, e.g.
+     * when a parameter is sent via a form.
+     *
+     * @return bool Whether $value is a positive int.
+     */
+    public static function isPositiveInteger($value): bool
+    {
+
+        // Reject arrays and objects
+        if (!is_scalar($value)) {
+            return false;
+        }
+
+        // First clause checks that only digits are present
+        // Second clause ensures a positive value
+        return ctype_digit(strval($value)) && intval($value) > 0;
+    }
```

Comment:
@johnsaigle isnt it a bit confusing if you pass a string and it returns true ?

i.e. `'5'` is a positive integer but not an integer since its a string... 

you see what I mean ? might be confusing with `is_int ( mixed $var ) : bool`

Reply:
Yeah I do see what you mean. Maybe I could change the name to like `stringIsPositiveInteger` and have it only take a string as input?

Reply:
maybe or something like `containsPositiveInteger()` making it clear its not it's type ?

Reply:
Yeah but `contains` makes me think that if a positive integer exists anywhere in the input that function should return true. Technically `abc100def` contains a positive integer...

Reply:
maybe `valueIsPositiveInterger` or `equalPositiveInteger`?

I Approve pending the proper name for this function.

Reply:
I like `valueIsPositiveInteger`